### PR TITLE
PXP-510: [CLI] Develop the list-instances command

### PR DESCRIPTION
**Pull request by jk-gan**
## Summary

screenshot - 2023-05-16T170014 774

Ticket: [PXP-510](https://mindvalley.atlassian.net/browse/PXP-510)

### Added
- [x] Add `wukong application instances list` command
- [x] integrate with API

[PXP-510]: https://mindvalley.atlassian.net/browse/PXP-510?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/99#" title="Author self-approved">jk-gan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by onimsha — CHANGES_REQUESTED** [review]

File: src/commands/application/instances/list.rs
Comment (jk-gan) [reply src/commands/application/instances/list.rs:78]: oh ya, I missed that part

File: src/commands/application/instances/list.rs
Comment (onimsha) [inline src/commands/application/instances/list.rs:79]: since we're listing on all namespaces, i think the text `on namespace production` is redundant. We should remove it.

Comment (jk-gan) [reply src/commands/application/instances/list.rs:79]: got it

**Review by onimsha — COMMENTED** [review]
Looks like the command does not accept the `--application` flag . 

```bash
❯ cargo run -- application instances list --namespace staging --application mv-connect
    Finished dev [unoptimized + debuginfo] target(s) in 1.22s
     Running `target/debug/wukong application instances list --namespace staging --application mv-connect`
Checking your permission to connect to the remote instance...✅
Listing running instances of the application mv-platform...✅
┌──────────────────────────────────────────────────┬──────────────┬────────────────┐
│ INSTANCE-NAME                                    │ INSTANCE-IP  │ INSTANCE-READY │
├──────────────────────────────────────────────────┼──────────────┼────────────────┤
│ green@staging/mv-platform-blue-6b7b9bfdf7-fnjkm  │ 10.0.128.16  │ true           │
├──────────────────────────────────────────────────┼──────────────┼────────────────┤
│ green@staging/mv-platform-green-78cf688949-ghq22 │ 10.0.128.113 │ true           │
├──────────────────────────────────────────────────┼──────────────┼────────────────┤
│ green@staging/platform-db-migration-gv5qm        │ 10.0.128.97  │ false          │
└──────────────────────────────────────────────────┴──────────────┴────────────────┘
```

**Comment by jk-gan** [comment]
> Looks like the command does not accept the `--application` flag .
> 
> ```shell
> ❯ cargo run -- application instances list --namespace staging --application mv-connect
>     Finished dev [unoptimized + debuginfo] target(s) in 1.22s
>      Running `target/debug/wukong application instances list --namespace staging --application mv-connect`
> Checking your permission to connect to the remote instance...✅
> Listing running instances of the application mv-platform...✅
> ┌──────────────────────────────────────────────────┬──────────────┬────────────────┐
> │ INSTANCE-NAME                                    │ INSTANCE-IP  │ INSTANCE-READY │
> ├──────────────────────────────────────────────────┼──────────────┼────────────────┤
> │ green@staging/mv-platform-blue-6b7b9bfdf7-fnjkm  │ 10.0.128.16  │ true           │
> ├──────────────────────────────────────────────────┼──────────────┼────────────────┤
> │ green@staging/mv-platform-green-78cf688949-ghq22 │ 10.0.128.113 │ true           │
> ├──────────────────────────────────────────────────┼──────────────┼────────────────┤
> │ green@staging/platform-db-migration-gv5qm        │ 10.0.128.97  │ false          │
> └──────────────────────────────────────────────────┴──────────────┴────────────────┘
> ```

wait, it should use the `--application` value, let me check

**Comment by jk-gan** [comment]
> Looks like the command does not accept the `--application` flag .
> 
> ```shell
> ❯ cargo run -- application instances list --namespace staging --application mv-connect
>     Finished dev [unoptimized + debuginfo] target(s) in 1.22s
>      Running `target/debug/wukong application instances list --namespace staging --application mv-connect`
> Checking your permission to connect to the remote instance...✅
> Listing running instances of the application mv-platform...✅
> ┌──────────────────────────────────────────────────┬──────────────┬────────────────┐
> │ INSTANCE-NAME                                    │ INSTANCE-IP  │ INSTANCE-READY │
> ├──────────────────────────────────────────────────┼──────────────┼────────────────┤
> │ green@staging/mv-platform-blue-6b7b9bfdf7-fnjkm  │ 10.0.128.16  │ true           │
> ├──────────────────────────────────────────────────┼──────────────┼────────────────┤
> │ green@staging/mv-platform-green-78cf688949-ghq22 │ 10.0.128.113 │ true           │
> ├──────────────────────────────────────────────────┼──────────────┼────────────────┤
> │ green@staging/platform-db-migration-gv5qm        │ 10.0.128.97  │ false          │
> └──────────────────────────────────────────────────┴──────────────┴────────────────┘
> ```

Fixed.

**Review by onimsha — DISMISSED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/99#" title="Author self-approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/99#pullrequestreview-1438328153" title="Approved">onimsha</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/99#" title="Author self-approved">jk-gan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/99#" title="Author self-approved">jk-gan</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by onimsha — APPROVED** [review]

**Comment by sauron-droid** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/mindvalley/wukong-cli/pull/99#" title="Author self-approved">jk-gan</a>*, *<a href="https://github.com/mindvalley/wukong-cli/pull/99#pullrequestreview-1440831840" title="Approved">onimsha</a>*


The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=mindvalley%2Fwukong-cli).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/mindvalley/wukong-cli/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
